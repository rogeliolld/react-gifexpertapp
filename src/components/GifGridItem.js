import React from 'react'

export const GifGridItem = ({id, title, url}) => {
    
    return (
        <div className="card animate__animated animate__fadeInDown animate__delay-1s">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
