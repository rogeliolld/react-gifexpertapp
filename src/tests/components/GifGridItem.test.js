import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas <GifGridItem />", () => {
  const title = "Un titulo";
  const url = "https://localhost/algo.png";

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe tener un pÃ¡rrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("debe de tener la Class Css animate__fadeInDown", () => {
    const div = wrapper.find("div");

    const className = div.prop("className");

    expect(className.includes("animate__fadeInDown")).toBe(true);
  });
});
