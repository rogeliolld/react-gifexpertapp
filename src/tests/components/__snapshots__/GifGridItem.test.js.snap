// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown animate__delay-1s"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.png"
  />
  <p>
    Un titulo
  </p>
</div>
`;
